paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/settings/PlaygroundSettings.kt

```diff
@@ -136,38 +169,69 @@ internal class PlaygroundSettings private constructor(
         }
     }
 
+    @Serializable
+    private class SerializableSettings(
+        val configurationData: PlaygroundConfigurationData,
+        val settings: Map<String, String>,
+    )
+
     companion object {
         private const val sharedPreferencesName = "PlaygroundSettings"
         private const val sharedPreferencesKey = "json"
 
         fun createFromDefaults(): PlaygroundSettings {
+            val defaultConfigurationData = PlaygroundConfigurationData()
+
             val settings = allSettingDefinitions.associateWith { settingDefinition ->
-                MutableStateFlow(settingDefinition.defaultValue)
+                val defaultValue = settingDefinition.playgroundConfigurable()?.convertToValue(defaultConfigurationData)
+                    ?: settingDefinition.defaultValue
+
+                MutableStateFlow(defaultValue)
             }.toMutableMap()
-            return PlaygroundSettings(settings)
+            return PlaygroundSettings(defaultConfigurationData, settings)
         }
 
         fun createFromJsonString(jsonString: String): PlaygroundSettings {
-            val settings: MutableMap<PlaygroundSettingDefinition<*>, MutableStateFlow<Any?>> =
-                mutableMapOf()
-            val jsonObject = Json.decodeFromString(JsonObject.serializer(), jsonString)
+            val settings: MutableMap<PlaygroundSettingDefinition<*>, MutableStateFlow<Any?>> = mutableMapOf()
+
+            val unserializedSettings = try {
+                Json.decodeFromString(SerializableSettings.serializer(), jsonString)
+            } catch (exception: SerializationException) {
+                Log.d("PlaygroundSettingsError", exception.message ?: exception.toString())
+
+                // If fails, attempt backwards compatible approach
```

Comment:
Why would it fail?

Reply:
Previously, all the settings were root-level saved. This update moves them into a nested object called `settings` while the root object contains both the `settings` object and `configurationData` object. If we fail to serialize using the new method, we then attempt to serialize with the old method.

We can remove this migration code later after everyone has used the playground, keep it so that all our old QR codes work, or just have everyone reset their app data.

Reply:
As discussed on Slack, let's just reset the playground settings to default.